deps/build.jl

```diff
@@ -19,7 +19,7 @@ function prog_version(prog)
         return nothing
     end
     try
-       return convert(VersionNumber, v)
+       return VersionNumber(v)
```

Comment:
Is there reason not to add that code to Base for the `VersionNumber` constructor and/or the `@v_str` macro?

Reply:
My impression is that non-semver parsing is deemed out of scope for Base. By putting it in Base (in Julia 1.0) we would be committing to a particular nonstandard parsing scheme for a long time. The nice thing about having permissive parsing in a package is that it can be quickly updated when weird real-world examples arise. 

See e.g. https://github.com/JuliaLang/julia/issues/7282